app/mixins/form.js

```diff
@@ -54,7 +54,7 @@ export default Mixin.create({
       }
 
       let $form;
-      if (this.get('tagName') === 'form' || (this.$() && this.$().prop('tagName'))) {
+      if (this.get('tagName') === 'FORM' || (this.$() && this.$().prop('tagName') === 'FORM')) {
```

Comment:
Because the first one `this.get('tagName')` could be set by the user and can be either in lower or upper.